### DISCUSSION TITLE
Where can I find what versions of mysql are supported by `mysql2` package?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [sidorares/node-mysql2#1838] [Q&A]
Where can I find what versions of mysql are supported by `mysql2` package?
I'm having trouble connecting to my DB & was wondering if its because its I'm using MYSQL v 8.0

**Comment by sidorares** [comment]
The answer is "its complicated"

Most mysql servers respond "v10" as a version of mysql protocol, BUT
- a lot of features ( compression, ssl, connect attributes, type of auth handshake, support of auth plugins ) can be negotiated at connect time, this depends on both mysql server version and server config
- same for auth plugins, just a bit later at a handshake sequence

It would be good to compile the matrix of supoorted ( or partially supported ) features. I created an issue to track that - https://github.com/sidorares/node-mysql2/issues/1839

Can you elaborate a bit on "I'm having trouble connecting" - what errors you see, what have you tried etc so I can help solving the issue

**Reply by cliffordfajardo** [reply]
Hi @sidorares,
I've created a github issue with more details: https://github.com/sidorares/node-mysql2/issues/1840

Thanks for responding to my discussion and the additional link to better understand the mysql environment space; helpful for me as a longtime postgres user

**Comment by sidorares** [comment]
list of capabilities flags: https://dev.mysql.com/doc/dev/mysql-server/latest/group__group__cs__capabilities__flags.html
